### PR TITLE
fix: set error catch properly

### PR DESCRIPTION
**Pull request by netanelC**
Set a case when the given error from S3 is not from type Error. This PR makes sure that the error will be from type Error

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: tests/unit/fileSyncerManager/fileSyncerManager.spec.ts [hunk 1]
```diff
@@ -60,6 +60,20 @@ describe('fileSyncerManager', () => {
       expect(providerManagerMock.dest.postFile).toHaveBeenCalled();
     });
 
+    it('When found a task with index not 0, it starts from the index', async () => {
```
Comment (RonitKissis) [inline tests/unit/fileSyncerManager/fileSyncerManager.spec.ts]: change to -1

File: tests/unit/fileSyncerManager/fileSyncerManager.spec.ts [hunk 2]
```diff
@@ -70,6 +84,16 @@ describe('fileSyncerManager', () => {
       expect(taskHandlerMock.reject).toHaveBeenCalled();
     });
 
+    it(`When get or post file throws unknown error, catches the error`, async () => {
```
Comment (RonitKissis) [inline tests/unit/fileSyncerManager/fileSyncerManager.spec.ts:87]: typo- catch anot catches

File: tests/unit/fileSyncerManager/fileSyncerManager.spec.ts (hunk 1 above)
Comment (netanelC) [reply tests/unit/fileSyncerManager/fileSyncerManager.spec.ts]: Done!

File: tests/unit/fileSyncerManager/fileSyncerManager.spec.ts (hunk 2 above)
Comment (netanelC) [reply tests/unit/fileSyncerManager/fileSyncerManager.spec.ts:87]: Why? third body, it is supposed to be "catches"

**Review by RonitKissis — APPROVED** [review]
amazing job! MEZUDA...!

**Review by RonitKissis — APPROVED** [review]
amazing job! MEZUDA...!

**Pull request merged**